Python/pywarpx/_libwarpx.py

```diff
@@ -397,9 +398,10 @@ def add_particles(species_name, x=0., y=0., z=0., ux=0., uy=0., uz=0.,
     lenux = np.size(ux)
     lenuy = np.size(uy)
     lenuz = np.size(uz)
+    lenw = np.size(w)
 
     if (lenx == 0 or leny == 0 or lenz == 0 or lenux == 0 or
```

Comment:
I wonder if we should make those `and` combined, to avoid (silent) returns where one parameter is only set to zero?
We have a check for mismatched lengths below.

Reply:
cc @dpgrote , what do you think?

Reply:
This is tricky because of the default value of parameters. If a user calls `add_particles(electrons, x=newx)` each step to inject particles and on some steps the length of `newx` is zero, it should return silently. With the `or` this is what happens. Using `and` and relying on the length consistency check below would raise an error since the length of the default parameters is 1 and so `maxlen` would be 1.

Perhaps the default values should be changed to `None` with the argument handling taking that into account. With that it is known which arguments were supplied by the user and the consistency of those can be checked. This would require a fair amount of reworking of this code.

Reply:
Seeing as the default arguments for all of those attributes are `0.` i.e. has `np.size(--) == 1`, a user would have to actually pass an empty array to this function for the above condition to be triggered. Why should we not just handle that case similarly as the situation where an attribute is simply not specified and set it to an array of length `maxlen` with all zeros?

Reply:
I see. I think my confusion just lies into what happens with unprovided parameters. Do we default init those to zero or so?

Reply:
Yes, parameters not supplied default to `0.`, which has a length of 1. If multiple particles are being added, the zero is broadcast over all of the particles.

Reply:
If an input parameter has a length of zero, using the default value instead is probably not the correct thing to do. In my example above, if `newx` has a length of zero, using the default value would erroneously add a particle with `x=0.`, not the expected result.

Reply:
Ok, I think I understand now. Yep, I agree that we should do a follow-up to use `=None` as default parameter to make things more clear/Pythonic.

Reply:
I can do that after this PR merges.